rerun_cpp/src/rerun/time_column.hpp

```diff
@@ -0,0 +1,143 @@
+            SortingStatus sorting_status = SortingStatus::Sorted
+        ) {
+            return TimeColumn(
+                Timeline(timeline_name, TimeType::Sequence),
+                sequence_points,
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from an array of sequence points.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param timepoints_in_nanoseconds The time points in nanoseconds.
+        /// Make sure the sorting status is correctly specified.
+        /// \param sorting_status The sorting status of the time points.
+        static TimeColumn from_times_nanoseconds(
+            std::string_view timeline_name, Collection<int64_t> timepoints_in_nanoseconds,
+            SortingStatus sorting_status = SortingStatus::Sorted
+        ) {
+            return TimeColumn(
+                Timeline(timeline_name, TimeType::Time),
+                timepoints_in_nanoseconds,
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from an array of arbitrary std::chrono durations.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param chrono_timepoints The time points as chrono durations.
+        /// Make sure the sorting status is correctly specified.
+        /// \param sorting_status The sorting status of the time points.
+        template <typename TRep, typename TPeriod>
+        static TimeColumn from_times(
+            std::string_view timeline_name,
+            const Collection<std::chrono::duration<TRep, TPeriod>>& chrono_timepoints,
+            SortingStatus sorting_status = SortingStatus::Sorted
+        ) {
+            std::vector<int64_t> timepoints(chrono_timepoints.size());
+            for (size_t i = 0; i < chrono_timepoints.size(); i++) {
+                timepoints[i] =
+                    std::chrono::duration_cast<std::chrono::nanoseconds>(chrono_timepoints[i])
+                        .count();
+            }
+            return TimeColumn(
+                Timeline(timeline_name, TimeType::Time),
+                std::move(timepoints),
+                sorting_status
+            );
+        }
+
+        /// Creates a sequence time column from a range of sequence points.
+        ///
+        /// \param timeline_name The name of the timeline this column belongs to.
+        /// \param min The minimum sequence point, must be less than `max`.
+        /// \param max The maximum sequence point, must be greater than `min`.
+        /// \param step The step size between sequence points. Must be non-zero..
+        static TimeColumn from_sequence_range(
+            std::string_view timeline_name, int64_t min, int64_t max, int64_t step = 1
+        ) {
+            assert(step > 0);
+            assert(min < max);
```

Comment:
```suggestion
            assert(min <= max);
```

Reply:
Hmmm I figured it doesn't make any sense to have min==max, but yeah I guess why not, there could be a single element!

Reply:
but then I also have to allow step size==0.
I'm sticking with `min<max`!

Reply:
No, I think it is VERY important to allow logging an empty column batch!

The number of plot points may be out of the hands of the coder, and so they just hand it off to Rerun, and it all works, until one day there are no plot points, and they get a runtime assert.